content/en/blog/_posts/2023-05-12-in-place-pod-resize/index.md

```diff
@@ -0,0 +1,200 @@
+value of `InProgress` indicates that the node has accepted the resize request
+and is in the process of applying the resize request to the pod's containers.
+A value of `Deferred` means that the requested resize cannot be granted at this
+time, and the node will keep retrying. The resize may be granted when other pods
+leave and free up node resources. A value of `Infeasible` is a signal that the
+node cannot accommodate the requested resize. This can happen if the requested
+resize exceeds the maximum resources the node can ever allocate for a pod.
+
+
+## When to use this feature
+
+Here are a few examples where this feature may be useful:
+- Pod is running on node but with either too much or too little resources.
+- Pods are not being scheduled do to lack of sufficient CPU or memory in a
+cluster that is underutilized by running pods that were overprovisioned.
+- Evicting certain stateful pods that need more resources to schedule them
+on bigger nodes is an expensive or disruptive operation when other lower
+priority pods in the node can be resized down or moved.
+
+
+## How to use this feature
+
+In order to use this feature in v1.27, the `InPlacePodVerticalScaling`
+feature gate must be enabled. A local cluster with this feature enabled
+can be started as shown below:
+
+```
+root@vbuild:~/go/src/k8s.io/kubernetes# FEATURE_GATES=InPlacePodVerticalScaling=true ./hack/local-up-cluster.sh
+go version go1.20.2 linux/arm64
++++ [0320 13:52:02] Building go targets for linux/arm64
+    k8s.io/kubernetes/cmd/kubectl (static)
+    k8s.io/kubernetes/cmd/kube-apiserver (static)
+    k8s.io/kubernetes/cmd/kube-controller-manager (static)
+    k8s.io/kubernetes/cmd/cloud-controller-manager (non-static)
+    k8s.io/kubernetes/cmd/kubelet (non-static)
+...
+...
+Logs:
+  /tmp/etcd.log
+  /tmp/kube-apiserver.log
+  /tmp/kube-controller-manager.log
+
+  /tmp/kube-proxy.log
+  /tmp/kube-scheduler.log
+  /tmp/kubelet.log
+
+To start using your cluster, you can open up another terminal/tab and run:
+
+  export KUBECONFIG=/var/run/kubernetes/admin.kubeconfig
+  cluster/kubectl.sh
+
+Alternatively, you can write to the default kubeconfig:
+
+  export KUBERNETES_PROVIDER=local
+
+  cluster/kubectl.sh config set-cluster local --server=https://localhost:6443 --certificate-authority=/var/run/kubernetes/server-ca.crt
+  cluster/kubectl.sh config set-credentials myself --client-key=/var/run/kubernetes/client-admin.key --client-certificate=/var/run/kubernetes/client-admin.crt
+  cluster/kubectl.sh config set-context local --cluster=local --user=myself
+  cluster/kubectl.sh config use-context local
+  cluster/kubectl.sh
+
+```
+
+Once the local cluster is up and running, Kubernetes users can schedule pods
+with resources, and resize the pods via kubectl. An example of how to use this
+feature is illustrated in the following demo video.
+
+[![](in-place-pod-resize-demo.png)](https://youtu.be/1m2FOuB6Bh0)
```

Comment:
For accessibility, use the video title as the alt attribute. You can also use the [`youtube`](https://gohugo.io/content-management/shortcodes/#youtube) shortcode (which I recommend).

Reply:
I don't have a way of previewing/verifying what it would look like in md editor. Does this look right?

```{{< youtube id="1m2FOuB6Bh0" title="In-place resize of pod CPU and memory resources">}}```

Reply:
You can [preview the website locally](https://kubernetes.io/docs/contribute/new-content/open-a-pr/#preview-locally).